packages/ember-glimmer/tests/integration/helpers/element-action-test.js

```diff
@@ -1489,4 +1489,47 @@ moduleFor('Helpers test: element action', class extends RenderingTest {
     this.assert.ok(outerActionCalled, 'the action fired on the proper target');
     this.assert.ok(innerClickCalled, 'the click was triggered');
   }
+
+  ['@glimmer element action with (mut undefinedThing) works properly']() {
+    let component;
+
+    let ExampleComponent = Component.extend({
+      member: undefined,
+      init() {
+        this._super(...arguments);
+        component = this;
+      }
+    });
+
+    this.registerComponent('example-component', {
+      ComponentClass: ExampleComponent,
+      template: '<button {{action (mut member) "foo"}}>Click me</button>'
+    });
+
+    this.render('{{example-component}}');
+
+    this.assert.equal(component.get('member'), undefined, 'value is initially undefined');
+
+    this.runTask(() => {
+      this.$('button').click();
+    });
+
+    this.assert.equal(component.get('member'), 'foo', 'value is updated');
+
+    this.runTask(() => {
+      component.set('member', 'b');
+    });
+
+    this.assert.equal(component.get('member'), 'b', 'precond - value is updated');
+
+    this.runTask(() => {
+      this.$('button').click();
+    });
+
+    this.assert.equal(component.get('member'), 'foo', 'value is updated');
+
+    this.runTask(() => {
+      component.set('member', undefined);
+    });
```

Comment:
No, the failure scenario to start with was just having `{{action (mut someUndefinedValue)}}`. Setting back to `undefined` with a rerender is the "reset" step.